pandas/core/arrays/base.py

```diff
@@ -293,8 +298,16 @@ def _from_factorized(cls, values, original):
     # Must be a Sequence
     # ------------------------------------------------------------------------
 
+    @overload
+    def __getitem__(self, item: int) -> Any:
+        ...
+
+    @overload
+    def __getitem__(self, item: Union[slice, np.ndarray, List[Any]]) -> ExtensionArray:
```

Comment:
Ellipsis, Tuple[any_of_the_above]

Reply:
Not sure what you mean here @jbrockmendel

Reply:
could also get `arr[..., :]`

Reply:
I don't think that is possible in an extension array because they are one dimensional.   Can you illustrate an example?

Reply:
there is a PR that has already had some review ( that I was planning to reopen once we had numpy types but won't be necessary if added here) #37921 so that may help.

Reply:
@simonjayhawkins That PR has the following for the `__getitem__` signatures:
```python
    @overload
    def __getitem__(self, item: int) -> EAScalarOrMissing:  # type: ignore[misc]
        ...

    @overload
    def __getitem__(self, item: Union[slice, np.ndarray]) -> ExtensionArray:
        ...

    def __getitem__(
        self, item: Union[int, slice, np.ndarray]
    ) -> Union[ExtensionArray, EAScalarOrMissing]:
```

So what I have added is more expansive.  Are you suggesting there is more to do here beyond what I've already done?

Reply:
> So what I have added is more expansive.

sorry didn't mean to suggest that my PR was correct (otherwise it would have been merged!). just that we have had some previous discussion that may (or may not) help.

Reply:
> > So what I have added is more expansive.
> 
> sorry didn't mean to suggest that my PR was correct (otherwise it would have been merged!). just that we have had some previous discussion that may (or may not) help.

I took a look and didn't see anything there. Some discussion on what to do in the 2D case

Reply:
> I don't think that is possible in an extension array because they are one dimensional. Can you illustrate an example?

Several EA subclasses can be 2D

Reply:
and even when 1D, `arr[..., :N]` should resolve to `arr[:N]

Reply:
I removed the `List[Any]` in a recent commit, so we can decide on whether to expand the possible arguments to `__getitem__()` at some other time.